Makefile

```diff
@@ -51,3 +51,27 @@ publish.codeclimate: ## Publish coverage to Code Climate
 		./cc-test-reporter sum-coverage --output 'coverage/total.json' coverage/javascript.json coverage/rspec.json
 		# Upload the our test coverage to code climate
 		./cc-test-reporter upload-coverage --input coverage/total.json
+
+.PHONY: qa
+qa: ## Set DEPLOY_ENV to qa
+	$(eval DEPLOY_ENV=qa)
+
+.PHONY: staging
+staging: ## Set DEPLOY_ENV to staging
+	$(eval DEPLOY_ENV=staging)
+
+.PHONY: production
+production: ## Set DEPLOY_ENV to production
+	$(eval DEPLOY_ENV=production)
+
+.PHONY: plan
+plan: qa ## Run terraform for ${DEPLOY_ENV} eg: make qa plan, make staging plan, make production plan
+	$(eval export TF_VAR_paas_app_secrets_file=terraform/workspace_variables/app_secrets.yml)
+	$(eval export TF_VAR_paas_app_config_file=terraform/workspace_variables/app_config.yml)
+	terraform init -backend-config terraform/workspace_variables/${DEPLOY_ENV}_backend.tfvars terraform
+	terraform plan -var-file terraform/workspace_variables/${DEPLOY_ENV}.tfvars -out terraform/workspace_variables/${DEPLOY_ENV}.tfplan terraform
+	echo 'Run => "make ${DEPLOY_ENV} deploy" to deploy this plan'
+
+.PHONY: deploy
+deploy: ## Run terraform for ${DEPLOY_ENV} eg: make qa deploy, make staging deploy, make production deploy
+	terraform apply "terraform/workspace_variables/${DEPLOY_ENV}.tfplan"
```

Comment:
This means deploy is dependent on plan. Can we use a simple `terraform apply` instead?
And you don't need to output the tf plan in `plan`